build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -22,6 +22,13 @@ part 'serialization.dart';
 
 /// All the [AssetId]s involved in a build, and all of their outputs.
 class AssetGraph {
+  /// A map of phase number to primary inputs that failed during that phase.
+  ///
+  /// See [markActionFailed] and [markActionSucceeded] for more info.
+  final Map<int, Set<AssetId>> _failedActions;
+  UnmodifiableMapView<int, Iterable<AssetId>> get failedActions =>
```

Comment:
why do we need to expose this as `UnmodifiableMapView` instead of `Map`?

Reply:
I wasn't clear...

It makes sense to me to wrap it in `UnmodifiableMapView` for defensiveness.. I'm not sure why the getter's signature also needs to say that.

Reply:
Oh it probably doesn't I guess, it makes it a bit more clear that its not modifiable though?

Reply:
The interface is the same, the methods you can't use will still autocomplete in your editor etc. It's only if you hover over it and look for the type that you'd notice.

Reply:
> It's only if you hover over it and look for the type that you'd notice.

Ya but that is a fairly common use case right? At least that is usually how I figure out the type of a thing.

Why wouldn't we want to expose this type?

Reply:
🤷‍♂️ 

I'm fine with it if you think it adds value.